File: source/back-end/src/main/java/be/pxl/backend/service/IProductService.java
package be.pxl.backend.service;

import java.util.List;
import be.pxl.backend.entity.Product;

public interface IProductService {

	public Product find(int id);
	
	public List<Product> all();
	
	public void persist(Product product);
	
	public void delete(int id);
	
	public void update(int id, Product product);
	
}

File: Source/frontend/frontend.Service/IProductService.cs
using frontend.Domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace frontend.Service
{
    public interface IProductService
    {
        List<Product> All();
        Product Find(int id);
        void Add(Product product);
        void Update(Product product);
        void Delete(int id);
        void ChangeStatus(Product product);
    }
}
